ServerCore/Pages/Events/FastestSolves.cshtml.cs

```diff
@@ -15,12 +16,24 @@ public class FastestSolvesModel : EventSpecificPageModel
     {
         public List<PuzzleStats> Puzzles { get; private set; }
 
+        public PuzzleStateFilter? StateFilter { get; set; }
+
+        public SortOrder? Sort { get; set; }
+
+        public Team CurrentTeam { get; set; }
+
+        private const SortOrder DefaultSort = SortOrder.RankAscending;
+
         public FastestSolvesModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
         {
         }
 
-        public async Task OnGetAsync()
+        public async Task OnGetAsync(SortOrder? sort, PuzzleStateFilter? stateFilter)
         {
+            this.Sort = sort;
+            this.StateFilter = stateFilter;
+            this.CurrentTeam = (await UserEventHelper.GetTeamForPlayer(_context, Event, LoggedInUser));
```

Comment:
CurrentTeam should be null when the EventRole is not Player. If I'm looking at status as an admin it should not be filtered by my team's progress.